HiveMQ Edge JDK HttpServer Exception

## Expected behavior
HiveMQ Edge server starts normally.

## Actual behavior
HiveMQ Edge servers starts with a JDK HttpServer ProcessingException error

## To Reproduce
Unzip the downloaded HiveMQ Edge v2024.1 program to C:\hivemq on a Windows 10 computer.  Download and install the OpenJDK11 JRE from https://adoptium.net/?variant=openjdk11.  Right click on the run.bat file in the \bin directory and select "Run as administrator".  

### Steps

### Reproducer code

## Details
I suspect the web GUI server is trying to use a port that is already in use, but the port HiveMQ Edge is attempting to use isn't listed.  I read through the wiki, and didn't see any configuration setting for how to set the web GUI port.  The default web GUI port isn't documented anywhere in the wiki.  I've included the error message displayed below.


2024-02-09 09:53:43,051 INFO  - Started MQTT TCP Listener on address 0.0.0.0 and on port 1883
2024-02-09 09:53:44,188 ERROR - Uncaught Error:
javax.ws.rs.ProcessingException: IOException thrown when creating the JDK HttpServer.
	at org.glassfish.jersey.jdkhttp.JdkHttpServerFactory.createHttpServer(JdkHttpServerFactory.java:232)
	at org.glassfish.jersey.jdkhttp.JdkHttpServerFactory.createHttpServer(JdkHttpServerFactory.java:176)
	at org.glassfish.jersey.jdkhttp.JdkHttpServerFactory.createHttpServer(JdkHttpServerFactory.java:84)
	at com.hivemq.http.JaxrsHttpServer.startServer(JaxrsHttpServer.java:136)
	at com.hivemq.HiveMQEdgeMain.startApiServer(HiveMQEdgeMain.java:205)
	at com.hivemq.HiveMQEdgeMain.startGateway(HiveMQEdgeMain.java:169)
	at com.hivemq.HiveMQEdgeMain.start(HiveMQEdgeMain.java:224)
	at com.hivemq.HiveMQEdgeMain.main(HiveMQEdgeMain.java:244)
	Caused by: java.net.BindException: Address already in use: bind
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Unknown Source)
	at java.base/sun.nio.ch.Net.bind(Unknown Source)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(Unknown Source)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(Unknown Source)
	at jdk.httpserver/sun.net.httpserver.ServerImpl.<init>(Unknown Source)
	at jdk.httpserver/sun.net.httpserver.HttpServerImpl.<init>(Unknown Source)
	at jdk.httpserver/sun.net.httpserver.DefaultHttpServerProvider.createHttpServer(Unknown Source)
	at jdk.httpserver/com.sun.net.httpserver.HttpServer.create(Unknown Source)
	at org.glassfish.jersey.jdkhttp.JdkHttpServerFactory.createHttpServer(JdkHttpServerFactory.java:229)
	... 7 common frames omitted

- Affected HiveMQ Edge version(s):  2024.1
- Used JVM version:  11.0.22

--- a/hivemq-edge/src/main/java/com/hivemq/http/JaxrsHttpServer.java
+++ b/hivemq-edge/src/main/java/com/hivemq/http/JaxrsHttpServer.java
@@ -38,6 +38,7 @@ import org.glassfish.jersey.server.ResourceConfig;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.ws.rs.ProcessingException;
 import javax.ws.rs.core.UriBuilder;
 import javax.ws.rs.ext.ExceptionMapper;
 import java.net.URI;
@@ -62,7 +63,6 @@ public class JaxrsHttpServer {
     private final @NotNull List<HttpServer> httpServers = new ArrayList<>();
     private final @NotNull List<JaxrsHttpServerConfiguration> configs;
     private final @Nullable ResourceConfig resourceConfig;
-    private @Nullable ShutdownHooks shutdownHooks = null;
     private @Nullable JaxrsObjectMapperProvider objectMapperProvider;
     private final @NotNull Object mutex = new Object();
     private volatile boolean running = false;
@@ -73,14 +73,7 @@ public class JaxrsHttpServer {
             final @Nullable ResourceConfig resourceConfig) {
         this.configs = configs;
         this.resourceConfig = resourceConfig;
-        this.shutdownHooks = shutdownHooks;
-        this.shutdownHooks.add(new Shutdown());
-    }
-
-    public JaxrsHttpServer(
-            final @NotNull List<JaxrsHttpServerConfiguration> configs, final @Nullable ResourceConfig resourceConfig) {
-        this.configs = configs;
-        this.resourceConfig = resourceConfig;
+        shutdownHooks.add(new Shutdown());
     }
 
     public void startServer() {
@@ -124,18 +117,36 @@ public class JaxrsHttpServer {
                         if (isSecure) {
                             Preconditions.checkNotNull(config.getSslContext(),
                                     "When configured with SSL, context must be supplied");
-                            httpServer = JdkHttpServerFactory.createHttpServer(baseUri,
-                                    resources,
-                                    config.getSslContext(),
-                                    false);
-                            if (config.getHttpsConfigurator() != null) {
-                                logger.debug("WebServer configured with SSL [{}] configuration..",
-                                        config.getHttpsConfigurator().getSSLContext().getProtocol());
-                                ((HttpsServer) httpServer).setHttpsConfigurator(config.getHttpsConfigurator());
+                            try {
+                                httpServer = JdkHttpServerFactory.createHttpServer(baseUri,
+                                        resources,
+                                        config.getSslContext(),
+                                        false);
+                                if (config.getHttpsConfigurator() != null) {
+                                    logger.debug("WebServer configured with SSL [{}] configuration..",
+                                            config.getHttpsConfigurator().getSSLContext().getProtocol());
+                                    ((HttpsServer) httpServer).setHttpsConfigurator(config.getHttpsConfigurator());
+                                }
+                            } catch (ProcessingException processingException) {
+                                logger.error(
+                                        "Unable to start the Https Server for uri '{}'. Is another service using the port '{}'?",
+                                        baseUri,
+                                        config.getPort());
+                                // still throw the exception to mitigate a silent fail
+                                throw processingException;
                             }
                         } else {
                             logger.trace("Creating HTTP service {} with {}", baseUri, resources);
-                            httpServer = JdkHttpServerFactory.createHttpServer(baseUri, resources, false);
+                            try {
+                                httpServer = JdkHttpServerFactory.createHttpServer(baseUri, resources, false);
+                            } catch (ProcessingException processingException) {
+                                logger.error(
+                                        "Unable to start the Http Server for uri '{}'. Is another service using the port '{}'?",
+                                        baseUri,
+                                        config.getPort());
+                                // still throw the exception to mitigate a silent fail
+                                throw processingException;
+                            }
                         }
                         logger.trace("Created API server in {}ms", (System.currentTimeMillis() - startCreate));
                         final long startRegister = System.currentTimeMillis();
@@ -147,9 +158,16 @@ public class JaxrsHttpServer {
                         //-- If a static resource path has been supplied in config, ensure we mount it
                         registerStaticRoot(config, httpServer);
 
-                        registerContext("/app", new WebAppHandler(objectMapperProvider.getMapper(), "httpd"), httpServer);
-                        registerContext("/images", new StaticFileHandler(objectMapperProvider.getMapper(), "httpd/images"), httpServer);
-                        registerContext("/module/images", new AlternativeClassloadingStaticFileHandler(objectMapperProvider.getMapper(), "httpd/images"), httpServer);
+                        registerContext("/app",
+                                new WebAppHandler(objectMapperProvider.getMapper(), "httpd"),
+                                httpServer);
+                        registerContext("/images",
+                                new StaticFileHandler(objectMapperProvider.getMapper(), "httpd/images"),
+                                httpServer);
+                        registerContext("/module/images",
+                                new AlternativeClassloadingStaticFileHandler(objectMapperProvider.getMapper(),
+                                        "httpd/images"),
+                                httpServer);
 
 
                         httpServers.add(httpServer);

--- a/hivemq-edge/src/main/java/com/hivemq/http/JaxrsHttpServer.java
+++ b/hivemq-edge/src/main/java/com/hivemq/http/JaxrsHttpServer.java
@@ -41,6 +41,7 @@ import org.slf4j.LoggerFactory;
 import javax.ws.rs.ProcessingException;
 import javax.ws.rs.core.UriBuilder;
 import javax.ws.rs.ext.ExceptionMapper;
+import java.net.BindException;
 import java.net.URI;
 import java.util.ArrayList;
 import java.util.List;
@@ -128,10 +129,17 @@ public class JaxrsHttpServer {
                                     ((HttpsServer) httpServer).setHttpsConfigurator(config.getHttpsConfigurator());
                                 }
                             } catch (ProcessingException processingException) {
-                                logger.error(
-                                        "Unable to start the Https Server for uri '{}'. Is another service using the port '{}'?",
-                                        baseUri,
-                                        config.getPort());
+                                if (processingException.getCause() instanceof BindException) {
+                                    logger.error(
+                                            "Unable to start the Http Server for uri '{}'. The port '{}' is already in use.",
+                                            baseUri,
+                                            config.getPort());
+                                } else {
+                                    logger.error(
+                                            "Unable to start the Http Server for uri '{}'. Is another service using the port '{}'?",
+                                            baseUri,
+                                            config.getPort());
+                                }
                                 // still throw the exception to mitigate a silent fail
                                 throw processingException;
                             }
@@ -140,10 +148,17 @@ public class JaxrsHttpServer {
                             try {
                                 httpServer = JdkHttpServerFactory.createHttpServer(baseUri, resources, false);
                             } catch (ProcessingException processingException) {
-                                logger.error(
-                                        "Unable to start the Http Server for uri '{}'. Is another service using the port '{}'?",
-                                        baseUri,
-                                        config.getPort());
+                                if (processingException.getCause() instanceof BindException) {
+                                    logger.error(
+                                            "Unable to start the Https Server for uri '{}'. The port '{}' is already in use.",
+                                            baseUri,
+                                            config.getPort());
+                                } else {
+                                    logger.error(
+                                            "Unable to start the Https Server for uri '{}'. Is another service using the port '{}'?",
+                                            baseUri,
+                                            config.getPort());
+                                }
                                 // still throw the exception to mitigate a silent fail
                                 throw processingException;
                             }

--- a/hivemq-edge/src/test/java/com/hivemq/api/JaxrsResourceTests.java
+++ b/hivemq-edge/src/test/java/com/hivemq/api/JaxrsResourceTests.java
@@ -34,6 +34,8 @@ import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
 
+import static org.mockito.Mockito.mock;
+
 /**
  * @author Simon L Johnson
  */
@@ -61,7 +63,7 @@ public class JaxrsResourceTests {
         //-- ensure we supplied our own test mapper as this can effect output
         ObjectMapper mapper = new ObjectMapper();
         config.setObjectMapper(mapper);
-        server = new JaxrsHttpServer(List.of(config), null);
+        server = new JaxrsHttpServer(mock(), List.of(config), null);
         server.startServer();
     }
 

--- a/hivemq-edge/src/test/java/com/hivemq/api/JaxrsSSLTests.java
+++ b/hivemq-edge/src/test/java/com/hivemq/api/JaxrsSSLTests.java
@@ -16,7 +16,7 @@
 package com.hivemq.api;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.hivemq.bootstrap.ioc.Injector;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
 import com.hivemq.http.JaxrsHttpServer;
 import com.hivemq.http.config.JaxrsHttpServerConfiguration;
 import com.hivemq.http.core.HttpResponse;
@@ -27,7 +27,6 @@ import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
-import org.mockito.Mock;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import util.TestKeyStoreGenerator;
@@ -47,6 +46,8 @@ import java.security.KeyStore;
 import java.security.SecureRandom;
 import java.util.List;
 
+import static org.mockito.Mockito.mock;
+
 /**
  * @author Simon L Johnson
  */
@@ -59,15 +60,12 @@ public class JaxrsSSLTests {
     static final int READ_TIMEOUT = 1000;
     static final String HTTPS = "https";
 
-    protected JaxrsHttpServer server;
-    @Mock
-    private Injector injector;
-    protected TestKeyStoreGenerator testKeyStoreGenerator;
-    protected SSLContext context;
+    protected @NotNull JaxrsHttpServer server;
+    protected @NotNull TestKeyStoreGenerator testKeyStoreGenerator;
+    protected @NotNull SSLContext context;
 
     @Before
     public void setUp() throws Exception {
-
         testKeyStoreGenerator = new TestKeyStoreGenerator();
         JaxrsHttpServerConfiguration config = new JaxrsHttpServerConfiguration();
         config.setPort(TEST_HTTP_PORT);
@@ -77,7 +75,7 @@ public class JaxrsSSLTests {
         config.addResourceClasses(TestApiResource.class);
         config.setHttpsConfigurator(new HttpsConfigurator(context) {
             @Override
-            public void configure(final HttpsParameters params) {
+            public void configure(final @NotNull HttpsParameters params) {
                 SSLParameters parameters = getSSLContext().getDefaultSSLParameters();
                 parameters.setProtocols(new String[]{"TLSv1.2"});
                 params.setSSLParameters(parameters);
@@ -86,7 +84,7 @@ public class JaxrsSSLTests {
         //-- ensure we supplied our own test mapper as this can effect output
         ObjectMapper mapper = new ObjectMapper();
         config.setObjectMapper(mapper);
-        server = new JaxrsHttpServer(List.of(config), null);
+        server = new JaxrsHttpServer(mock(), List.of(config), null);
         server.startServer();
     }
 
@@ -96,7 +94,7 @@ public class JaxrsSSLTests {
         testKeyStoreGenerator.release();
     }
 
-    private SSLContext getSslContext(String password) throws Exception {
+    private SSLContext getSslContext(final @NotNull String password) throws Exception {
         final KeyStore keyStore = createKeyStore(password);
         final KeyManagerFactory keyManagerFactory =
                 KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
@@ -108,34 +106,32 @@ public class JaxrsSSLTests {
         return sslContext;
     }
 
-    private TrustManagerFactory defaultTrustManager(KeyStore keyStore) throws GeneralSecurityException {
+    private TrustManagerFactory defaultTrustManager(@NotNull KeyStore keyStore) throws GeneralSecurityException {
         TrustManagerFactory trustManagerFactory =
                 TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
         trustManagerFactory.init(keyStore);
         return trustManagerFactory;
     }
 
-    private KeyStore createKeyStore(String password) throws Exception {
+    private KeyStore createKeyStore(@NotNull String password) throws Exception {
         KeyStore keyStore = KeyStore.getInstance("JKS");
         FileInputStream fis = new FileInputStream(generateKeystore(password));
         keyStore.load(fis, password.toCharArray());
         return keyStore;
     }
 
-    private KeyManager[] getKeyManagers(final KeyManagerFactory keyManagerFactory) {
-        final KeyManager[] keyManagers = keyManagerFactory.getKeyManagers();
-        return keyManagers;
+    private KeyManager @NotNull [] getKeyManagers(final KeyManagerFactory keyManagerFactory) {
+        return keyManagerFactory.getKeyManagers();
     }
 
-    protected File generateKeystore(String password) throws Exception {
+    protected @NotNull File generateKeystore(@NotNull String password) throws Exception {
         final File file = testKeyStoreGenerator.generateKeyStore("teststore", "JKS", password, password);
         file.deleteOnExit();
         return file;
     }
 
-    protected static String getTestServerAddress(String protocol, int port, String uri) {
-        String url = String.format("%s://%s:%s/%s", protocol, "localhost", port, uri);
-        return url;
+    protected static @NotNull String getTestServerAddress(@NotNull String protocol, int port, @NotNull String uri) {
+        return String.format("%s://%s:%s/%s", protocol, "localhost", port, uri);
     }
 
     @Test

--- a/hivemq-edge/src/test/java/com/hivemq/api/auth/BasicAuthenticationTests.java
+++ b/hivemq-edge/src/test/java/com/hivemq/api/auth/BasicAuthenticationTests.java
@@ -44,6 +44,8 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+import static org.mockito.Mockito.mock;
+
 /**
  * @author Simon L Johnson
  */
@@ -78,7 +80,7 @@ public class BasicAuthenticationTests {
         //-- ensure we supplied our own test mapper as this can effect output
         ObjectMapper mapper = new ObjectMapper();
         config.setObjectMapper(mapper);
-        server = new JaxrsHttpServer(List.of(config), conf);
+        server = new JaxrsHttpServer(mock(), List.of(config), conf);
         server.startServer();
     }
 

--- a/hivemq-edge/src/test/java/com/hivemq/api/auth/BearerTokenAuthTests.java
+++ b/hivemq-edge/src/test/java/com/hivemq/api/auth/BearerTokenAuthTests.java
@@ -52,6 +52,8 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+import static org.mockito.Mockito.mock;
+
 /**
  * @author Simon L Johnson
  */
@@ -92,7 +94,7 @@ public class BearerTokenAuthTests {
         //-- ensure we supplied our own test mapper as this can effect output
         ObjectMapper mapper = new ObjectMapper();
         config.setObjectMapper(mapper);
-        server = new JaxrsHttpServer(List.of(config), conf);
+        server = new JaxrsHttpServer(mock(), List.of(config), conf);
         server.startServer();
     }
 

--- a/hivemq-edge/src/test/java/com/hivemq/api/auth/ChainedAuthTests.java
+++ b/hivemq-edge/src/test/java/com/hivemq/api/auth/ChainedAuthTests.java
@@ -30,6 +30,7 @@ import com.hivemq.api.model.ApiErrorMessage;
 import com.hivemq.api.model.auth.ApiBearerToken;
 import com.hivemq.api.model.auth.UsernamePasswordCredentials;
 import com.hivemq.api.resources.impl.AuthenticationResourceImpl;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
 import com.hivemq.http.JaxrsHttpServer;
 import com.hivemq.http.config.JaxrsHttpServerConfiguration;
 import com.hivemq.http.core.HttpConstants;
@@ -52,6 +53,8 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+import static org.mockito.Mockito.mock;
+
 /**
  * @author Simon L Johnson
  */
@@ -64,14 +67,14 @@ public class ChainedAuthTests {
     static final int READ_TIMEOUT = 1000;
     static final String HTTP = "http";
 
-    protected static JaxrsHttpServer server;
+    protected static @NotNull JaxrsHttpServer server;
 
     @BeforeClass
     public static void setUp() throws Exception {
         JaxrsHttpServerConfiguration config = new JaxrsHttpServerConfiguration();
         config.setPort(TEST_HTTP_PORT);
 
-        ApiJwtConfiguration configuration = new ApiJwtConfiguration(2048, "Test-Issuer","Test-Audience", 10, 2);
+        ApiJwtConfiguration configuration = new ApiJwtConfiguration(2048, "Test-Issuer", "Test-Audience", 10, 2);
         JwtAuthenticationProvider jwtAuthenticationProvider = new JwtAuthenticationProvider(configuration);
 
         IUsernamePasswordProvider usernamePasswordProvider = AuthTestUtils.createTestUsernamePasswordProvider();
@@ -79,7 +82,8 @@ public class ChainedAuthTests {
         authenticationHandlers.add(new BearerTokenAuthenticationHandler(jwtAuthenticationProvider));
         authenticationHandlers.add(new BasicAuthenticationHandler(usernamePasswordProvider));
 
-        ResourceConfig conf = new ResourceConfig(){{
+        ResourceConfig conf = new ResourceConfig() {
+            {
                 register(new ApiAuthenticationFeature(authenticationHandlers));
             }
         };
@@ -87,20 +91,21 @@ public class ChainedAuthTests {
         conf.register(TestPermitAllApiResource.class);
         conf.register(TestResourceLevelRolesApiResource.class);
         conf.register(new AuthenticationResourceImpl(usernamePasswordProvider,
-                jwtAuthenticationProvider, jwtAuthenticationProvider));
+                jwtAuthenticationProvider,
+                jwtAuthenticationProvider));
         //-- ensure we supplied our own test mapper as this can effect output
         ObjectMapper mapper = new ObjectMapper();
         config.setObjectMapper(mapper);
-        server = new JaxrsHttpServer(List.of(config), conf);
+        server = new JaxrsHttpServer(mock(), List.of(config), conf);
         server.startServer();
     }
 
     @AfterClass
-    public static void tearDown(){
+    public static void tearDown() {
         server.stopServer();
     }
 
-    protected static String getTestServerAddress(String protocol, int port, String uri){
+    protected static String getTestServerAddress(final @NotNull String protocol, int port, String uri) {
         String url = String.format("%s://%s:%s/%s", protocol, "localhost", port, uri);
         return url;
     }
@@ -110,13 +115,15 @@ public class ChainedAuthTests {
 
         ObjectMapper mapper = new ObjectMapper();
         UsernamePasswordCredentials creds = new UsernamePasswordCredentials("testuser", "test");
-        HttpResponse response =
-                HttpUrlConnectionClient.post(HttpUrlConnectionClient.JSON_HEADERS,
-                        getTestServerAddress(HTTP, TEST_HTTP_PORT, "api/v1/auth/authenticate"),
-                        new ByteArrayInputStream(mapper.writeValueAsBytes(creds)),
-                        CONNECT_TIMEOUT, READ_TIMEOUT);
+        HttpResponse response = HttpUrlConnectionClient.post(HttpUrlConnectionClient.JSON_HEADERS,
+                getTestServerAddress(HTTP, TEST_HTTP_PORT, "api/v1/auth/authenticate"),
+                new ByteArrayInputStream(mapper.writeValueAsBytes(creds)),
+                CONNECT_TIMEOUT,
+                READ_TIMEOUT);
         Assert.assertEquals("Resource should be accepted", 200, response.getStatusCode());
-        Assert.assertEquals("API authenticate response should be json", MediaType.APPLICATION_JSON, response.getContentType());
+        Assert.assertEquals("API authenticate response should be json",
+                MediaType.APPLICATION_JSON,
+                response.getContentType());
         ApiBearerToken token = mapper.readValue(response.getResponseBody(), ApiBearerToken.class);
         Assert.assertNotNull("Response should contain a bearer token", token.getToken());
     }
@@ -126,15 +133,19 @@ public class ChainedAuthTests {
 
         ObjectMapper mapper = new ObjectMapper();
         UsernamePasswordCredentials creds = new UsernamePasswordCredentials("testuser", "invalidpassword");
-        HttpResponse response =
-                HttpUrlConnectionClient.post(HttpUrlConnectionClient.JSON_HEADERS,
-                        getTestServerAddress(HTTP, TEST_HTTP_PORT, "api/v1/auth/authenticate"),
-                        new ByteArrayInputStream(mapper.writeValueAsBytes(creds)),
-                        CONNECT_TIMEOUT, READ_TIMEOUT);
+        HttpResponse response = HttpUrlConnectionClient.post(HttpUrlConnectionClient.JSON_HEADERS,
+                getTestServerAddress(HTTP, TEST_HTTP_PORT, "api/v1/auth/authenticate"),
+                new ByteArrayInputStream(mapper.writeValueAsBytes(creds)),
+                CONNECT_TIMEOUT,
+                READ_TIMEOUT);
         Assert.assertEquals("Resource should NOT be accepted", 401, response.getStatusCode());
-        Assert.assertEquals("API authenticate response should be json", MediaType.APPLICATION_JSON, response.getContentType());
+        Assert.assertEquals("API authenticate response should be json",
+                MediaType.APPLICATION_JSON,
+                response.getContentType());
         ApiErrorMessage message = mapper.readValue(response.getResponseBody(), ApiErrorMessage.class);
-        Assert.assertEquals("Response should indicate correct failure message", "Invalid username and/or password", message.getTitle());
+        Assert.assertEquals("Response should indicate correct failure message",
+                "Invalid username and/or password",
+                message.getTitle());
     }
 
     @Test
@@ -142,34 +153,39 @@ public class ChainedAuthTests {
 
         ObjectMapper mapper = new ObjectMapper();
         UsernamePasswordCredentials creds = new UsernamePasswordCredentials("testuser", "test");
-        HttpResponse response =
-                HttpUrlConnectionClient.post(HttpUrlConnectionClient.JSON_HEADERS,
-                        getTestServerAddress(HTTP, TEST_HTTP_PORT, "api/v1/auth/authenticate"),
-                        new ByteArrayInputStream(mapper.writeValueAsBytes(creds)),
-                        CONNECT_TIMEOUT, READ_TIMEOUT);
+        HttpResponse response = HttpUrlConnectionClient.post(HttpUrlConnectionClient.JSON_HEADERS,
+                getTestServerAddress(HTTP, TEST_HTTP_PORT, "api/v1/auth/authenticate"),
+                new ByteArrayInputStream(mapper.writeValueAsBytes(creds)),
+                CONNECT_TIMEOUT,
+                READ_TIMEOUT);
         Assert.assertEquals("Resource should be accepted", 200, response.getStatusCode());
-        Assert.assertEquals("API authenticate response should be json", MediaType.APPLICATION_JSON, response.getContentType());
+        Assert.assertEquals("API authenticate response should be json",
+                MediaType.APPLICATION_JSON,
+                response.getContentType());
         ApiBearerToken token = mapper.readValue(response.getResponseBody(), ApiBearerToken.class);
         Assert.assertNotNull("Response should contain a bearer token", token.getToken());
 
         //-- now validate the token against the UNSECURE API which returns whether its valid
-        response =
-                HttpUrlConnectionClient.post(HttpUrlConnectionClient.JSON_HEADERS,
-                        getTestServerAddress(HTTP, TEST_HTTP_PORT, "api/v1/auth/validate-token"),
-                        new ByteArrayInputStream(mapper.writeValueAsBytes(token)),
-                        CONNECT_TIMEOUT, READ_TIMEOUT);
+        response = HttpUrlConnectionClient.post(HttpUrlConnectionClient.JSON_HEADERS,
+                getTestServerAddress(HTTP, TEST_HTTP_PORT, "api/v1/auth/validate-token"),
+                new ByteArrayInputStream(mapper.writeValueAsBytes(token)),
+                CONNECT_TIMEOUT,
+                READ_TIMEOUT);
         Assert.assertEquals("Resource should be accepted", 200, response.getStatusCode());
 
         //-- finally use it as a bear token header against a secure endpoint
 
         Map<String, String> headers = Map.of(HttpConstants.AUTH_HEADER,
-                HttpUtils.getBearerTokenAuthenticationHeaderValue(token.getToken()), "Content-Type", "application/json",
-                "Accept", "application/json");
+                HttpUtils.getBearerTokenAuthenticationHeaderValue(token.getToken()),
+                "Content-Type",
+                "application/json",
+                "Accept",
+                "application/json");
 
-        response =
-                HttpUrlConnectionClient.get(headers,
-                        getTestServerAddress(HTTP, TEST_HTTP_PORT, "test/get/auth/user"),
-                        CONNECT_TIMEOUT, READ_TIMEOUT);
+        response = HttpUrlConnectionClient.get(headers,
+                getTestServerAddress(HTTP, TEST_HTTP_PORT, "test/get/auth/user"),
+                CONNECT_TIMEOUT,
+                READ_TIMEOUT);
         Assert.assertEquals("Resource should be accepted", 200, response.getStatusCode());
         ApiPrincipal user = mapper.readValue(response.getResponseBody(), ApiPrincipal.class);
         Assert.assertEquals("Username should match that supplied at point of auth", "testuser", user.getName());
@@ -180,11 +196,12 @@ public class ChainedAuthTests {
     public void testGetSecuredResourceWithBasicAuthHeader() throws IOException {
 
         ObjectMapper mapper = new ObjectMapper();
-        Map<String, String> headers = Map.of(HttpConstants.AUTH_HEADER,
-                HttpUtils.getBasicAuthenticationHeaderValue("testadmin", "test"));
-        HttpResponse response =
-                HttpUrlConnectionClient.get(headers,
-                        getTestServerAddress(HTTP, TEST_HTTP_PORT, "test/get/auth/admin"), CONNECT_TIMEOUT, READ_TIMEOUT);
+        Map<String, String> headers =
+                Map.of(HttpConstants.AUTH_HEADER, HttpUtils.getBasicAuthenticationHeaderValue("testadmin", "test"));
+        HttpResponse response = HttpUrlConnectionClient.get(headers,
+                getTestServerAddress(HTTP, TEST_HTTP_PORT, "test/get/auth/admin"),
+                CONNECT_TIMEOUT,
+                READ_TIMEOUT);
         Assert.assertEquals("Resource should be accepted", 200, response.getStatusCode());
         ApiPrincipal user = mapper.readValue(response.getResponseBody(), ApiPrincipal.class);
         Assert.assertEquals("Username should match that supplied at point of auth", "testadmin", user.getName());

